applications/FluidDynamicsApplication/python_scripts/navier_stokes_two_fluids_solver.py

```diff
@@ -254,75 +254,73 @@ def InitializeSolutionStep(self):
+        self.main_model_part.ProcessInfo.SetValue(KratosCFD.VOLUME_ERROR, volume_error)
 
-            # We set this value at every time step as other processes/solvers also use them
-            dynamic_tau = self.settings["formulation"]["dynamic_tau"].GetDouble()
-            self.main_model_part.ProcessInfo.SetValue(KratosMultiphysics.DYNAMIC_TAU, dynamic_tau)
+        # We set this value at every time step as other processes/solvers also use them
+        dynamic_tau = self.settings["formulation"]["dynamic_tau"].GetDouble()
+        self.main_model_part.ProcessInfo.SetValue(KratosMultiphysics.DYNAMIC_TAU, dynamic_tau)
 
 
     def FinalizeSolutionStep(self):
         KratosMultiphysics.Logger.PrintInfo(self.__class__.__name__, "Mass and momentum conservation equations are solved.")
 
-        if self._TimeBufferIsInitialized():
-            # Recompute the distance field according to the new level-set position
-            if self._reinitialization_type != "none":
-                self._GetDistanceReinitializationProcess().Execute()
-                KratosMultiphysics.Logger.PrintInfo(self.__class__.__name__, "Redistancing process is finished.")
-
-            # Prepare distance correction for next step
-            self._GetDistanceModificationProcess().ExecuteFinalizeSolutionStep()
-
-            # Finalize the solver current step
-            self._GetSolutionStrategy().FinalizeSolutionStep()
-            # Limit the obtained acceleration for the next step
-            # This limitation should be called on the second solution step onwards (e.g. STEP=3 for BDF2)
-            # We intentionally avoid correcting the acceleration in the first resolution step as this might cause problems with zero initial conditions
-            if self._apply_acceleration_limitation and self.main_model_part.ProcessInfo[KratosMultiphysics.STEP] >= self.min_buffer_size:
-                self._GetAccelerationLimitationUtility().Execute()
+        # Recompute the distance field according to the new level-set position
+        if self._reinitialization_type != "none":
+            self._GetDistanceReinitializationProcess().Execute()
+            KratosMultiphysics.Logger.PrintInfo(self.__class__.__name__, "Redistancing process is finished.")
+
+        # Prepare distance correction for next step
+        self._GetDistanceModificationProcess().ExecuteFinalizeSolutionStep()
+
+        # Finalize the solver current step
+        self._GetSolutionStrategy().FinalizeSolutionStep()
+        # Limit the obtained acceleration for the next step
+        # This limitation should be called on the second solution step onwards (e.g. STEP=3 for BDF2)
+        # We intentionally avoid correcting the acceleration in the first resolution step as this might cause problems with zero initial conditions
+        if self._apply_acceleration_limitation and self.main_model_part.ProcessInfo[KratosMultiphysics.STEP] >= self.min_buffer_size:
```

Comment:
I think this should be now
`if self._apply_acceleration_limitation and self.main_model_part.ProcessInfo[KratosMultiphysics.STEP] > 1:`

right? So it is only applied after the first time step is fully solved.

Reply:
You're right. Thanks for spotting!

Reply:
Done!